packages/ts/form/src/Models.ts

```diff
@@ -34,57 +21,38 @@ export function hasFromString<T>(model: AbstractModel<T>): model is AbstractMode
   return _fromString in model;
 }
 
-export interface HasValue<T> {
-  value?: T;
-}
+export type Value<M> = M extends AbstractModel<infer T> ? T : never;
+
+export const _modelDetachedParent = { $value: undefined };
```

Comment:
Let's not use underscore for anything except Symbols.

```suggestion
export const modelDetachedParent = { $value$: undefined };
```